tests/run-make/arm-target-cpu-fp/rmake.rs

```diff
@@ -0,0 +1,37 @@
+// This tests that target-cpu correctly enables additional floating-point features.
+
+use run_make_support::{llvm_filecheck, llvm_objdump, rustc, static_lib_name};
+
+struct TestCase {
+    target: &'static str,
+    cpu: &'static str,
+}
+
+static CASES: &[TestCase] = &[TestCase { target: "armv8r-none-eabihf", cpu: "cortex-r52" }];
+
+fn main() {
+    for case in CASES {
+        let lib = static_lib_name(case.cpu);
+
+        rustc()
+            .edition("2021")
+            .arg(format!("--target={}", case.target))
+            .arg(format!("-Ctarget-cpu={}", case.cpu))
+            .arg("-Copt-level=3")
+            .crate_type("rlib")
+            .input("lib.rs")
+            .output(&lib)
+            .run();
+
+        let dis = llvm_objdump()
+            .arg("--arch-name=arm")
+            .arg(format!("--mcpu={}", case.cpu))
+            .disassemble()
+            .input(&lib)
+            .run()
+            .stdout_utf8();
+
+        let check_file = format!("{}_{}.checks", case.target, case.cpu).replace("-", "_");
```

Comment:
hmm, is there a reason we have to use `.replace` here, btw?

Reply:
I was following the convention that `.checks` files in other rmake tests have underscores, but the target and cpu names have dashes. Is there a better way to achieve this?

Reply:
...I'm not sure why they do lmao.

Reply:
oh there's only one rmake test with `.checks` files.

Reply:
@chrisnc despite having multiple .checks files that's actually a convention of 1, it's not actually convened yet. :^)

Reply:
Ack. Hyphens it is.

Reply:
Updated. Just using the cpu name now, as that uniquely identifies the architecture anyway...